api/catalog/api/views/image_views.py

```diff
@@ -132,7 +132,18 @@ def watermark(self, request, *_, **__):
         params.is_valid(raise_exception=True)
 
         image = self.get_object()
-        image_url = image.url
+
+        # initialising the image url to None by default
+        image_url = None
+
+        # checking if thumbnail url is available or not
+        # if yes , setting image_url to thumbnailurl
+        # if no , set image_url to its default url
```

Comment:
This comment isn't very helpful as it only repeats what the code below is saying. See rule#1 of the [best practices for writing code comments](https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/).

A better comment would be to explain what `image.thumbnail` is and why it might be messing it some cases.